apps/src/labs/ProjectManagerFactory.ts

```diff
@@ -0,0 +1,35 @@
+import {AppOptionsStore} from './AppOptionsStore';
+import {S3ChannelsStore, LocalChannelsStore} from './ChannelsStore';
+import ProjectManager from './ProjectManager';
+import {S3SourcesStore, LocalSourcesStore} from './SourcesStore';
+import {AppOptions, Project} from './types';
+
+export default class ProjectManagerFactory {
```

Comment:
From factory patterns I remember, the factory would take a parameter indicating which object type to instantiate, and would return the right specialization of the common base class, so that the caller could be agnostic about what it was getting back.  Would that logic be better here than in Music Lab?

Reply:
good call out, I did a pretty loose factory pattern here. I will update to be the more traditional pattern, which would be something like `getProjectManager(projectManagerType)`. Music Lab would still need to know the type it is requesting, so it would call the method similarly to how it's using the factory now.

Reply:
Good call out, I didn't quite do a factory here. One of my motivations here was to extract out the knowledge that `channelId` was part of `appOptions`, but given we are going to move it out of `appOptions` soon anyway I will update this. I'll make it a single method that gets called with the same parameters for local or s3. This is still useful because Music lab doesn't have to worry about creating sources or channels stores.